src/coreComponents/dataRepository/Group.hpp

```diff
@@ -1329,6 +1329,14 @@ class Group
   virtual void postRestartInitialization()
   {}
 
+  /**
+   * @brief Performs re-initialization of certain variable depending on the solver being used.
+   */
+  virtual void reinit()
+  {}
```

Comment:
Why do you have to add a member function to nearly all the classes of `GEOSX` when only the `WaveSolver*` family use them (I think)? Is it because of technical consideration in `python`?

Reply:
Yes we had a discussion with Randy and Josh about how we should re-initialize the variables from python, and it has been decided to create a `reini()` method that would be overriden in each Solver depending on the variables subject to re-initialization. I only override it the WaveSolverBase because I don't know how the other Solver should rewrite this function since I don't know how the other solvers work

Reply:
I guess I could declare this method in `SolverBase` though.

Reply:
OK. But why don't you implement a protected `reinit` member function only in `SolverBase` then?
Why do you have to put it on top in `Group` if you only expect `SolverBase` implementations to use it?

Reply:
I need it to be `Public` because the `reinit()` is called from `PySolver` files.

Reply:
I do not know all the details (specially the way you do in `python`) but in `C++`, if you explicitly call a `WaveSolver*` instance, you could have a `public` `reinit` member in `WaveSolver*` classes and `protected` (even `private` I think) in `SolverBase`.

Reply:
Ok I see what you mean, I can make it `Protected` in `SolverBase` and `Public` in `WaveSolver` then, it should be sufficient.

Reply:
Actually not, I really need it to be `Public` in `SolverBase` because I am manipulating `Solverbase` Object in `PySolver.cpp` when calling `reinit()`